test/run_test.py

```diff
@@ -464,6 +464,7 @@ def run_test(
     extra_unittest_args=None,
     env=None,
 ) -> int:
+    cache_dir = tempfile.TemporaryDirectory()
```

Comment:
This might makes Zain's work to save and restore cache directories across PR's jobs trickier IMO as this is now become a temp file instead of the local `.pytest_cache` directory

Reply:
yeah, but not really sure how to keep different processes from overriding each others caches, im also a bit surprised we havent run into this problem before, so maybe they dont really need different directories?

Reply:
changed so that --sc takes in a key/path so that they dont collide and we can use the normal cache directory again